service/dap/server.go

```diff
@@ -113,6 +113,8 @@ type Server struct {
 	exceptionErr error
 	// clientCapabilities tracks special settings for handling debug session requests.
 	clientCapabilities dapClientCapabilites
+	// logMessages maps a breakpoint id to a log message
+	logMessages map[int]string
```

Comment:
IMHO this is going to be a pain to keep in sync with breakpoint deletion, it's probably better to put it in proc.Breakpoint.

Reply:
I like the idea of having the log message attached to proc.Breakpoint, since its simpler.

One issue with adding this to proc.Breakpoint would be that we would have to store a format to store the log message in. If we store it on the breakpoint I think it would make sense to store it as a string, but we would have to parse it every time.

Storing it in the server would allow us to parse it when setting the breakpoint and storing it in a way that we could easily evaluate the necessary expressions without having to re-parse the message (we could swap string out for a format string and a list of expressions to evaluate).

Reply:
I'd be ok with having a `UserData interface{}` field in proc.Breakpoint.